d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/StateUpdater.java

```diff
@@ -318,13 +318,18 @@ else if (trackerClientState.getHealthScore() < MAX_HEALTH_SCORE
         {
           trackerClientState.setHealthState(TrackerClientState.HealthState.NEUTRAL);
         }
+
+        if (trackerClient.doNotLoadBalance())
```

Comment:
Can we skip the above health score logic (line 267-320) if the client has set doNotLoadBalancer? We can just keep HealthState.HEALTHY right?

Reply:
good catch, added

Reply:
I mean, we don't need to execute through line 278-230 if doNotLoadBalance is set, do we? Can we check the flag first at line 278
```
if (trackerClient.doNotLoadBalancer()) {
    // set health state and health score
} else {
    // normal health score adjustment logic
}
```

Reply:
Also have we agreed on not load balancing on error rate as well? Looks like we're turning off load balancing completely.

Reply:
We agreed to not load balance on error, no? 

I see your point, we can skip that logic.

Reply:
[our discussion ](https://docs.google.com/document/d/1C2IVFf74zjFO1H3Mgk0JGpmJumGjN7Rg2K23_gGhFOc/edit?disco=AAAAOAnK3uM) over the design doc was to load balance on error rate as the bottom line guard. But it seems you and Dyno team had some new updates.

Reply:
Added some info, btw I don't think we should include links to internal docs since this is a public repo.